src/ruleset.ts

```diff
@@ -775,24 +772,152 @@ export default {
       then: {
         function: pattern,
         functionOptions: {
-          match: "/^https:/",
+          notMatch: "/^http:/",
         },
       },
     },
 
     /**
-     * API8:2019 — Injection
+     * @author: Jason Harmon <https://github.com/jharmn>
+     */
+    "owasp:api8:2023-define-error-validation": {
+      message: "Missing error response of either 400, 422 or 4XX.",
+      description:
+        "Carefully define schemas for all the API responses, including either 400, 422 or 4XX responses which describe errors caused by invalid requests.",
+      severity: DiagnosticSeverity.Warning,
+      given: "$.paths..responses",
+      then: [
+        {
+          function: schema,
+          functionOptions: {
+            schema: {
+              type: "object",
+              anyOf: [
+                {
+                  required: ["400"],
+                },
+                {
+                  required: ["422"],
+                },
+                {
+                  required: ["4XX"],
+                },
+              ],
+            },
+          },
+        },
+      ],
+    },
+
+    /**
+     * @author: Jason Harmon <https://github.com/jharmn>
+     */
+    "owasp:api8:2023-define-error-responses-401": {
+      message: "Operation is missing {{property}}.",
+      description:
+        "OWASP API Security recommends defining schemas for all responses, even errors. The 401 describes what happens when a request is unauthorized, so its important to define this not just for documentation, but to empower contract testing to make sure the proper JSON structure is being returned instead of leaking implementation details in backtraces.",
+      severity: DiagnosticSeverity.Warning,
+      given: "$.paths..responses",
```

Comment:
I think we can make these JSON paths a bit more specific i.e. target various places that have responses property rather than doing `..responses`. This can lead to erroneous messages and be relatively slow. 

Don't need to do this in this PR but probably good to create an issue for this 🤔

Reply:
We need to be a little bit specific because there have been several issues popping up where JSON in examples, where the example JSON might be `{ responses: { ... } }` so I've always tried to point it somewhat in the right direction. I would love somebody better versed with nimma performance to help me understand the right balance of specific and fast but I don't know it insde and out.